dask/tests/test_config.py

```diff
@@ -195,15 +195,33 @@ def test_set():
 
     with set({'abc': 123}):
         assert config['abc'] == 123
+    assert 'abc' not in config
 
     with set({'abc.x': 1, 'abc.y': 2, 'abc.z.a': 3}):
         assert config['abc'] == {'x': 1, 'y': 2, 'z': {'a': 3}}
+    assert 'abc' not in config
 
     d = {}
     set({'abc.x': 123}, config=d)
     assert d['abc']['x'] == 123
 
 
+def test_set_nested():
+    with set({'abc': {'x': 123}}):
+        assert config['abc'] == {'x': 123}
+        with set({'abc.y': 456}):
+            assert config['abc'] == {'x': 123, 'y': 456}
+        assert config['abc'] == {'x': 123}
+    assert 'abc' not in config
```

Comment:
This test doesn't hit every case, but perhaps the whole test suite does. I haven't personally run it, but it'd be nice to have full coverage of `set` and `_assign`:

```
coverage run `which py.test` dask/tests/test_config.py`
coverage html
```

Reply:
We have full coverage on these functions

Reply:
The test above this one handles most of the testing coverage.